docs/docs/api/LayoutAnimations/KeyframeAnimations.md

```diff
@@ -0,0 +1,179 @@
+---
+id: keyframeAnimations
+title: Keyframe Animations
+sidebar_label: Keyframe Animations
+---
+
+The document explains how you can define complex animation using simple and popular animation definitions schema - Keyframes.
+
+## How to define Keyframe animation?
+
+### 1. Import Keyframe
+
+```js
+import { Keyframe } from 'react-native-reanimated';
+```
+
+### 2. Create Keyframe object, define initial and final state
+
+In Keyframe's constructor pass object with definitions of your animation. Object keys correspond to animation progress,
+so to 0 assign the style, you want for your object at the beginning of the animation and to 100 assign the style you want for your object to have at the end of the animation.
+
+```js
+import { Keyframe } from 'react-native-reanimated';
+
+const keyframe = new Keyframe({
+    0: {
+      transform: [ { rotate: '0deg' }],
+    },
+    100: {
+      transform: [ { rotate: '45deg' }],
+    },
+  }
+```
+
+### 3. Add middle points
+
+Between edge points, you can define middle points in which you want your object to have certain style properties.
+Remember that for every keyframe you should provide the same set of style properties!
```

Comment:
What happens if you omit them? I think better UX would be to preserve them with previous ones so using keyframes like:
```js
const keyframe = new Keyframe({
    0: {
      transform: [{ rotate: '0deg' }],
    },
    45: {
        transform: [{ transformX: 20 }]
    },
    100: {
      transform: [{ rotate: '45deg' }],
    },
  }
```

would keep rotation at 0 degrees on 45 key and add `transformX` that will be present on the last frame.

Reply:
It requires some changes in the parsing logic. I think preserving values with previous ones could be done quite easily. However, it will be harder to start animate new property in the middle of animation but I'm gonna think about it.

Reply:
Starting from a default value for a property is the one way to go here. It's easy for most props by using zero values with few exceptions like color or opacity. I think this is quite intuitive behavior. You'll need to traverse values first and merge a zero keyframe with props that are specified later on.

\cc @piaskowyk

Reply:
I had a similar idea. What if we require to pass all used props to zero keyframe as it contains initial values? That way we don't have to handle those edge cases with color, opacity, etc.

Reply:
I like this idea! We forgot about this before, but this approach is much better and more intuitive for users. Could we also add from-to syntax?
```js
const keyframe = new Keyframe({
  from: {
    transform: [{ rotate: '0deg' }],
  },
  to: {
    transform: [{ rotate: '45deg' }],
  },
});
```

Reply:
> What if we require to pass all used props to zero keyframe as it contains initial values? 

This is a great solution and we can relax this restriction in the future if we want to.